identity-service/src/routes/rewards.js

```diff
@@ -0,0 +1,41 @@
+const express = require('express')
+const { handleResponse, successResponse, errorResponseServerError } = require('../apiHelpers')
+const config = require('../config.js')
+const { RewardsReporter } = require('../utils/rewardsReporter')
+
+const rewardsRouter = express.Router()
+
+const handleResult = async ({ status, userId, challengeId, amount, error, phase, reporter, specifier }) => {
+  switch (status) {
+    case 'success':
+      await reporter.reportSuccess({ userId, challengeId, amount, specifier })
+      break
+    case 'failure':
+      await reporter.reportFailure({ userId, challengeId, amount, error, phase, specifier })
+      break
+    case 'rejection':
+      await reporter.reportAAORejection({ userId, challengeId, amount, error, specifier })
+      break
+    default:
+      throw new Error("Bad status code")
+  }
+}
+
+rewardsRouter.post('/attestation_result', handleResponse(async (req) => {
+  const { status, userId, challengeId, amount, error, phase, source, specifier } = req.body
+  const reporter = new RewardsReporter({
```

Comment:
Probably the reporter could be initialized as a singleton outside of the route itself

Reply:
Did that initially, but we use the source in initialization. It's a light weight object though so I think it's fine